tests/git_api.py

```diff
@@ -0,0 +1,45 @@
+import requests
+import json
+
+reposLIST = {'shipit': 'https://api.github.com/repos/mozilla-releng/ship-it/commits',
+            'services': 'https://api.github.com/repos/mozilla/release-services/commits',
+            'beetmoverscript': 'https://api.github.com/repos/mozilla-releng/beetmoverscript/commits',
+            'addonscript': 'https://api.github.com/repos/mozilla-releng/addonscript/commits',
+            'shipitv2': 'https://api.github.com/repos/mozilla-releng/shipit-v2/commits',
+            'build-cloud-tools': 'https://api.github.com/repos/mozilla-releng/build-cloud-tools/commits',
+            'build-puppet': 'https://api.github.com/repos/mozilla-releng/build-puppet/commits',
+            'shipitscript': 'https://api.github.com/repos/mozilla-releng/shipitscript/commits',
+            'bouncerscript': 'https://api.github.com/repos/mozilla-releng/bouncerscript/commits',
+            'treescript': 'https://api.github.com/repos/mozilla-releng/treescript/commits',
+            'mozapkpublisher': 'https://api.github.com/repos/mozilla-releng/mozapkpublisher/commits',
+            'OpenCloudConfig': 'https://api.github.com/repos/mozilla-releng/OpenCloudConfig/commits',
+            'scriptworker': 'https://api.github.com/repos/mozilla-releng/scriptworker/commits',
+            'pushsnapscript': 'https://api.github.com/repos/mozilla-releng/pushsnapscript/commits',
+            'signingscript': 'https://api.github.com/repos/mozilla-releng/signingscript/commits',
+            'cot-gpg-keys': 'https://api.github.com/repos/mozilla-releng/cot-gpg-keys/commits',
+            'pushapkscript': 'https://api.github.com/repos/mozilla-releng/pushapkscript/commits',
+            'balrogscript': 'https://api.github.com/repos/mozilla-releng/balrogscript/commits',
+            'funsize': 'https://api.github.com/repos/mozilla-releng/funsize/commits',
+            'signtool': 'https://api.github.com/repos/mozilla-releng/signtool/commits'
+}  
+for reposLIST_key in reposLIST:     # for loop to scroll through the reposLIST
+    r = requests.get(reposLIST.get(reposLIST_key))     # get infos from gitAPI page
+    p = r.json()     # turn into JSON content 
+    commit = {}     # dictionary with key = SHA and values=name, email, date, URL and message
+    for keys in p:    # loop to scroll through json content
+        author = {}    # dictionary with personal infor about commiter and commit
+        author.update({ 'Name: ' : keys['commit']['author']['name'], 
+                        'Email: ' : keys['commit']['author']['email'], 
+                        'Date: ' : keys['commit']['author']['date'], 
+                        'URL: ' : keys['commit']['url'], 
+                        'Message: ' : keys['commit']['message'] })     # add info in author dictionary
+        commit.update({ keys['sha'] : author })     # add info in commit dictionary
```

Comment:
I'm currently working on markdown for this json and  : commit.update({ keys['sha'] : author }) is making my life hard because for each commit there is another sha. I need to access what's stored in that sha dict. If it's possible , find a way to "have a more generic name for that dict". Maybe something like "Info" would work.

Reply:
Today I worked to fix the issue found by you, Hope to help
My .json  file contain three dictionaries ( reposLIST{},   commit{} and author{} ) structured as following:
reposLIST{ 'repo_name' :  commit{ 'sha' : author{ 'name' : '---', 'email' : '---', 'date' : '---', 'url' : '---', 'message' : '---'}}}   
If I understood correctly, your issue is when you want to select the information using 'sha' key. So, to fix it I switched the 'sha' key with the commit number.  

e.g repo_name -> 
-------------------0 -> name, date, url... 
-------------------1 -> name, date, url...
                                                                                                .
                                                                                                .
                                                                                                .
-------------------29 -> name, date, url...

Now you can access each commit using the commit number:
    e.g1:   
          for key in data:        ---# loop scroll through reposLIST dictionary
          ------print(data[key]['0']['Name: '])      ---#print the name from commit number 0 from each repository 
    e.g2:
           ------print(data['repo_name']['commit_number']['name' or 'email' or 'date' ...] ---#print the selected info from commit_number from repo_name


I can not use a generic name as a key because in that case every time the information will be overwritten and only the last commit will be saved.

I hope the above are useful.